filament/src/components/TransformManager.cpp

```diff
@@ -485,22 +490,18 @@ void FTransformManager::computeMaterialWorldOrientation(
     outOrientationCenter = parentOrientationCenter + localOrientationCenter;
 }
 
-math::mat3f FTransformManager::getMaterialCompoundOrientation(Instance ci) const noexcept {
+math::quatf FTransformManager::getMaterialCompoundOrientation(Instance ci) const noexcept {
     const mat4f& world = mManager[ci].world;
-    const mat3f& orientation = mManager[ci].materialOrientation;
+    const quatf& orientation = mManager[ci].materialOrientation;
 
-    // we need to strip everything from our world transform, except rotation
-    mat3f worldRotation = mat3f(world[0].xyz, world[1].xyz, world[2].xyz);
-    worldRotation[0] = normalize(worldRotation[0]);
-    worldRotation[1] = normalize(worldRotation[1]);
-    worldRotation[2] = normalize(worldRotation[2]);
+    const quatf worldRotationQuaternion = world.toQuaternion();
 
     // We need to apply the inverse of the world transformation's rotation component
-    // then apply our own orientation. Multiplying with worldRotation on the right is
-    // equivalent to multiplying on the left with the transpose of it, which in this
-    // case is equal to the inverse (orthonormal matrix). This allows us to compute only
-    // one matrix multiplication, instead of two (one on the left, one on the right).
-    return worldRotation * orientation;
+    // then apply our own orientation. Since the worldRotation is already applied to the material, 
```

Comment:
nit: You changed the name of the variable to worldRotationQuaternion, so should update it in the comment too